quinn-proto/src/crypto.rs

```diff
@@ -270,20 +273,25 @@ impl Crypto {
         self.write_nonce(&state, packet, nonce);
         let payload_len = payload.len();
 
-        let key = aead::OpeningKey::new(cipher, &state.key).unwrap();
         aead::open_in_place(&key, &*nonce, header, 0, payload.as_mut()).map_err(|_| ())?;
         payload.split_off(payload_len - cipher.tag_len());
         Ok(())
     }
 
     pub fn update(&self, side: Side) -> Crypto {
```

Comment:
Couldn't `CryptoContext::local` and `CryptoContext::remote` just be swapped in `CryptoContext::update` `CryptoState::update` use the same `cipher` and `digest` [here](https://github.com/djc/quinn/blob/f9a6935b2ffcda1f08e600b5ab4b87668965a5fc/quinn-proto/src/crypto.rs#L384) ?

Reply:
I don't understand this question; can you rephrase?

Reply:
I think you're asking if we can mutate the existing `Crypto` instead of instantiating a new one? In which case, we cannot, because we need to keep the old `Crypto` around for older packets that might still come in.